sycl/include/CL/sycl/half_type.hpp

```diff
@@ -36,24 +36,25 @@
 #endif
 
 #if __cpp_lib_bit_cast || __has_builtin(__builtin_bit_cast)
-#define __SYCL_CONSTEXPR_ON_HOST constexpr
+#define __SYCL_CONSTEXPR_HALF constexpr
 #else
-#define __SYCL_CONSTEXPR_ON_HOST
+#define __SYCL_CONSTEXPR_HALF 
 #endif
 
+#define __SYCL_DEVICE_OR_BITCAST_CONSTEXPR
 #ifdef __SYCL_DEVICE_ONLY__
-#define __SYCL_HALF_CTOR_CONSTEXPR constexpr
+#define __SYCL_DEVICE_OR_BITCAST_CONSTEXPR constexpr
 #elif __cpp_lib_bit_cast || __has_builtin(__builtin_bit_cast)
-#define __SYCL_HALF_CTOR_CONSTEXPR constexpr
+#define __SYCL_DEVICE_OR_BITCAST_CONSTEXPR constexpr
 #else
-#define __SYCL_HALF_CTOR_CONSTEXPR
+#define __SYCL_DEVICE_OR_BITCAST_CONSTEXPR
```

Comment:
BTW, why do we even need to distinguish between those two? Can we just leave `__SYCL_CONSTEXPR_HALF` and use it everywhere (i.e. in a way consistent between host and device)?

Reply:
Only __SYCL_CONSTEXPR_HALF macro is remained